core/src/main/java/io/questdb/griffin/engine/functions/groupby/ApproxMedianDoubleGroupByDefaultFunctionFactory.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.groupby;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.constants.DoubleConstant;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+
+public class ApproxMedianDoubleGroupByDefaultFunctionFactory implements FunctionFactory {
+    private final DoubleConstant percentileFunc = DoubleConstant.newInstance(0.5);
+
+    @Override
+    public String getSignature() {
+        return "approx_median(D)";
+    }
+
+    @Override
+    public boolean isGroupBy() {
+        return true;
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        return new ApproxPercentileDoubleGroupByFunction(args.getQuick(0), percentileFunc, ApproxPercentileLongGroupByDefaultFunctionFactory.DEFAULT_PRECISION, position);
```

Comment:
nit: this would read better with a static import (same applies to `ApproxMedianLongGroupByDefaultFunctionFactory`)
```suggestion
        return new ApproxPercentileDoubleGroupByFunction(args.getQuick(0), percentileFunc, DEFAULT_PRECISION, position);
```